comit/src/asset/bitcoin.rs

```diff
@@ -36,6 +37,54 @@ impl fmt::Display for Bitcoin {
     }
 }
 
+impl Serialize for Bitcoin {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let precision = 8;
+        let base: u64 = 10;
+        let remainder = self.as_sat() % base.pow(precision);
+        let integer = (self.as_sat() - remainder) / base.pow(precision);
+        let string = format!("{}.{}", integer, remainder);
+
+        serializer.serialize_str(&string)
+    }
+}
+
+impl<'de> Deserialize<'de> for Bitcoin {
+    /// Decimal precision of 8 is selected because it is the largest possible
+    /// precision for BTC
+    fn deserialize<D>(deserializer: D) -> Result<Self, <D as Deserializer<'de>>::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let precision = 8;
+        let string = String::deserialize(deserializer)?;
+        let v: Vec<&str> = string.as_str().split('.').collect();
+
+        let integer = *v.first().unwrap();
+        let decimals = *v.last().unwrap();
```

Comment:
These panics can easily be triggered by users if they supply a value that doesn't have a dot in it.

Just mentioning it as general feedback, I don't think we need these deserializers.